server/konnectors/linkedin.coffee

```diff
@@ -0,0 +1,328 @@
+            return next err if err
+
+            if body.status? and body.status == 'error'
+                next new Error(body.status_details)
+            datapoints = []
+
+            data = body.contact_data
+            datapoints = datapoints.concat getPhoneNumber(data)
+            datapoints = datapoints.concat getEmails(data)
+            datapoints = datapoints.concat getUrls(data)
+            datapoints = datapoints.concat getAddresses(data)
+
+
+            finalContact = new Contact
+                n: "#{data.last_name};#{data.first_name}"
+                fn: data.name
+                title: contact.title || undefined
+                org: data.company?.name || undefined
+                title: data.title || undefined
+                tags: ['linkedin']
+                datapoints: datapoints || undefined
+
+            finalContact.imageUrl = contact.secure_profile_image_url || undefined
+            ContactHelper.setAccount finalContact,
+                type: ACCOUNT_TYPE
+                name: entries.accountName
+                id: data.id
+
+            next null, finalContact
+
+
+    queue = async.queue processRetrievingContactData, 10
+
+    queue.drain = ->
+        log.info 'All data retrieved'
+        next()
+
+    Tag.getOrCreate { name: 'linkedin', color: '#1B86BC'}, (err, tag) ->
+        if err
+            entries.tag = null
+        else
+            entries.tag = tag
+        entries.listContacts.forEach (contact) ->
+            queue.push contact, (err, finalContact) ->
+                log.error err if err
+                saveContact finalContact, entries
+
+
+
+
+
+getPhoneNumber = (data) ->
+    listPhones = []
+
+    data.phone_numbers.forEach (number) ->
+        listPhones.push
+            name: 'tel'
+            type: number.type.toLowerCase()
+            value: number.number.replace(/ /g, '')
+    listPhones
+
+
+getEmails = (data) ->
+    listEmails = []
+
+    data.emails_extended.forEach (email) ->
+        listEmails.push
+            name: 'email'
+            value: email.email
+            type: 'internet'
+            pref: email.primary == true ? true : undefined
+    listEmails
+
+
+getUrls = (data) ->
+    listUrls = []
+
+    data.sites.forEach (site) ->
+        listUrls.push
+            name: 'url'
+            value: site.url
+            type: site.name
+    data.profiles.forEach (profile) ->
+        listUrls.push
+            name: 'url'
+            value: profile.url
+            type: 'linkedin'
+    data.twitter.forEach (twitter) ->
+        listUrls.push
+            name: 'url'
+            value: twitter.url
+            type: 'twitter'
+
+    listUrls
+
+getAddresses = (data) ->
+    listAddresses = []
+
+    if data.location?
+        segmentAddress = data.location.split(', ')
+            .reverse()
+        country = segmentAddress[0] || ''
+        #region = segmentAddress[1] || ''
+        #locality = segmentAddress[2] || ''
```

Comment:
@Peltoche OK, it's something that you should explain in a comment in your code ;-) The _why_ is often interesting and is difficult to infer from code.